homeassistant/components/islamic_prayer_times/sensor.py

```diff
@@ -16,36 +22,31 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 
     client = hass.data[DOMAIN]
 
-    dev = []
+    entities = []
     for sensor_type in SENSOR_TYPES:
-        dev.append(IslamicPrayerTimeSensor(sensor_type, client))
+        entities.append(IslamicPrayerTimeSensor(sensor_type, client))
 
-    async_add_entities(dev, True)
+    async_add_entities(entities, True)
 
 
 class IslamicPrayerTimeSensor(Entity):
     """Representation of an Islamic prayer time sensor."""
 
-    ENTITY_ID_FORMAT = "sensor.islamic_prayer_time_{}"
-
     def __init__(self, sensor_type, client):
         """Initialize the Islamic prayer time sensor."""
         self.sensor_type = sensor_type
-        self.entity_id = self.ENTITY_ID_FORMAT.format(self.sensor_type)
         self.client = client
-        self._name = self.sensor_type.capitalize()
-        self._state = None
-        self.unsub_update = None
+        self._state = self.client.prayer_times_info.get(self.sensor_type)
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return self._name
+        return f"{self.sensor_type} {SENSOR_SUFFIX}"
 
     @property
     def unique_id(self):
         """Return the unique id of the entity."""
-        return self.entity_id
+        return f"{DOMAIN}_{self.sensor_type}"
```

Comment:
We don't need to include the domain in the unique id. The entity registry is aware of both the integration domain and the platform domain.

Reply:
In this case I will set the sensor_type as unique_id.